pootle/apps/pootle_store/syncer.py

```diff
@@ -0,0 +1,363 @@
+    unit_sync_class = UnitSyncer
+
+    def __init__(self, store):
+        self.store = store
+
+    @cached_property
+    def disk_store(self):
+        return self.store.file.store
+
+    @property
+    def translation_project(self):
+        return self.store.translation_project
+
+    @property
+    def language(self):
+        return self.translation_project.language
+
+    @property
+    def project(self):
+        return self.translation_project.project
+
+    @property
+    def source_language(self):
+        return self.project.source_language
+
+    @property
+    def store_file_path(self):
+        return os.path.join(
+            self.translation_project.abs_real_path,
+            self.store.name)
+
+    @cached_property
+    def file_class(self):
+        # get a plugin adapted file_class
+        fileclass = format_classes.gather().get(
+            str(self.store.filetype.extension))
+        if fileclass:
+            return fileclass
+        if self.store.is_template:
+            # namedtuple is equiv here of object() with name attr
+            return self._getclass(
+                namedtuple("instance", "name")(
+                    name=".".join(
+                        [os.path.splitext(self.store.name)[0],
+                         str(self.store.filetype.extension)])))
+        return self._getclass(self.store)
+
+    def convert(self, fileclass=None):
+        """export to fileclass"""
+        fileclass = fileclass or self.file_class
+        logging.debug(
+            u"Converting %s to %s",
+            self.store.pootle_path,
+            fileclass)
+        output = fileclass()
+        output.settargetlanguage(self.language.code)
+        # FIXME: we should add some headers
+        for unit in self.store.units.iterator():
+            output.addunit(
+                self.unit_sync_class(unit).convert(output.UnitClass))
+        return output
+
+    def _getclass(self, obj):
+        try:
+            return getclass(obj)
+        except ValueError:
+            raise ValueError(
+                "Unable to find conversion class for Store '%s'"
+                % self.store.name)
+
+    def get_new_units(self, old_ids, new_ids):
+        return self.store.findid_bulk(
+            [self.store.dbid_index.get(uid)
+             for uid
+             in new_ids - old_ids])
+
+    def get_obsolete_units(self, old_ids, new_ids):
+        return (
+            self.disk_store.findid(uid)
+            for uid
+            in old_ids - new_ids)
+
+    def obsolete_unit(self, unit, conservative):
+        obsolete = 0
+        deleted = 0
+        del_unit = False
+        if not unit.istranslated():
+            del_unit = True
+        elif not conservative:
+            obsolete = 1
+            unit.makeobsolete()
+            if not unit.isobsolete():
+                del_unit = True
+        if del_unit:
+            deleted = 1
+            del unit
+        return obsolete, deleted
```

Comment:
and if a file format supports obsoleting then clashes are possible. because obsoleting and deleting can be different operations for the file format but pootle supports obsoleting only.